### ISSUE TITLE
Internal error: Missing error message or bug (v0.9.19)

### ISSUE THREAD

**Issue by jeremy-w** [idris-lang/Idris-dev#2685]
Given this input as file `order.idr`:

```
module Order

data Order: (left: Nat) -> (right: Nat) -> Type where
  lessThanOrEqual : Order right left
  greaterThanOrEqual : Order right left
```

This triggers an internal error:

```
*order> the Order 3 4 lessThanOrEqual
(input):1:5:When checking argument value to function Prelude.Basics.the:
        INTERNAL ERROR: Prelude.Nat.Nat -> Prelude.Nat.Nat -> Type ` is not a Type
```

Parenthesizing the `(Order 3 4)` bit works around this. If it weren't for the error message suggesting I file an issue, I would have just shrugged it off as user error.

Full interpreter session below:

```
> idris
     ____    __     _
    /  _/___/ /____(_)____
    / // __  / ___/ / ___/     Version 0.9.19
  _/ // /_/ / /  / (__  )      http://www.idris-lang.org/
 /___/\__,_/_/  /_/____/       Type :? for help

Idris is free software with ABSOLUTELY NO WARRANTY.
For details type :warranty.
Idris> :l order
Type checking ./order.idr
*order> :t Order
Order : Nat -> Nat -> Type
*order> lessThanOrEqual
(input):Can't infer argument right to lessThanOrEqual,
        Can't infer argument left to lessThanOrEqual
*order> the Order 3 4 lessThanOrEqual
(input):1:5:When checking argument value to function Prelude.Basics.the:
        INTERNAL ERROR: Prelude.Nat.Nat -> Prelude.Nat.Nat -> Type ` is not a Type
        This is probably a bug, or a missing error message.
        Please consider reporting at https://github.com/idris-lang/Idris-dev/issues
*order> the (Order 3 4) lessThanOrEqual
lessThanOrEqual : Order 3 4
```

**Comment by david-christiansen** [comment]
Thanks for a great report! Looks like we need to improve an error message :)

**Comment by raulfpl** [comment]
This seems to be fixed in idris 0.11.1.
When trying to reproduce the issue I got the expected error message:
```
(input):1:5:When checking argument a to function Prelude.Basics.the:
        Type mismatch between
                Nat -> Nat -> Type (Type of Order)
        and
                Type (Expected type)
```

**Comment by ahmadsalim** [comment]
Yes, this fixed now, I will close it. Thanks!
